etna/datasets/utils.py

```diff
@@ -175,3 +180,90 @@ def set_columns_wide(
     df_left.loc[timestamps_left_index, (segments_left_index, features_left_index)] = right_value.values
 
     return df_left
+
+
+def match_target_quantiles(features: Set[str]) -> Set[str]:
+    """Find quantiles in dataframe columns."""
+    pattern = re.compile("target_\d+\.\d+$")
+    return {i for i in list(features) if pattern.match(i) is not None}
+
+
+def get_target_with_quantiles(columns: pd.Index) -> Set[str]:
+    """Find "target" column and target quantiles among dataframe columns."""
+    column_names = set(columns.get_level_values(level="feature"))
+    target_columns = match_target_quantiles(column_names)
+    if "target" in column_names:
+        target_columns.add("target")
+    return target_columns
+
+
+def get_level_dataframe(
+    df: pd.DataFrame,
+    mapping_matrix: csr_matrix,
+    source_level_segments: Tuple[str, ...],
+    target_level_segments: Tuple[str, ...],
+):
+    """Perform mapping to dataframe at the target level.
+
+    Parameters
+    ----------
+    df:
+        dataframe at the source level
+    mapping_matrix:
+        mapping matrix between levels
+    source_level_segments:
+        tuple of segments at the source level
+    target_level_segments:
+        tuple of segments at the target level
+
+    Returns
+    -------
+    :
+       dataframe at the target level
+    """
+    target_names = tuple(get_target_with_quantiles(columns=df.columns))
+
+    if len(target_names) == 0:
+        raise ValueError("Provided dataframe has no columns with the target variable!")
+
+    df = df.loc[:, pd.IndexSlice[source_level_segments, target_names]]
+
+    if df.empty:
+        raise ValueError("Provided `source_level_segments` don't contain any valid segments!")
+
+    if len(set(df.columns.get_level_values(level="segment"))) != mapping_matrix.shape[1]:
```

Comment:
Gust use `source_level_segments`

Reply:
We can't just use `source_level_segments`, because it may contain redundant segement names.